### PR TITLE
feat: add migration script for storage v2

### PR DESCRIPTION
**Pull request by tanlethanh**

### CODE REVIEW

File: packages/store/migrations/index.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+) => {
+	logger.info(`start migrating database, scope: ${scope}`);
+	const setting = await storage.safeGet<SettingDocument>('settings');
+	if (!setting?.profile?.id) return;
+
+	const storedVersion = setting?.config?.storageVersion || 0;
+	const latestVersion = migrations[migrations.length - 1].version;
+
+	if (storedVersion < latestVersion) {
+		const newerFilter = (i: Migration) =>
```
Comment (FME849) [inline packages/store/migrations/index.ts:26]: I think `filterNewMigrations` or `filterVersionMigrations` have more transparent meaning

File: packages/store/migrations/index.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+						await storage.removeDoc(token._id);
+					});
+				});
+
+			const migrateNFTPromises = storage
+				.find<PouchDocument<unknown>>({
+					selector: { type: 'NFT', account: { $exists: true } },
+				})
+				.then(({ docs: tokens }) => {
+					return tokens.map(async (token) => {
```
Comment (FME849) [inline packages/store/migrations/index.ts:69]: do we need return value here?

File: packages/store/utils/type.ts [hunk 3]
```diff
@@ -19,6 +19,11 @@ import type {
...
 } from '@walless/core';
 
+import type helpers from '../plugins';
+
+export type Database = Omit<PouchDB.Database, 'find'> &
+	typeof helpers & { find: TypedFind };
+
 export type DocumentType =
 	| 'Setting'
 	// | 'EndpointMap'
```
Comment (FME849) [inline packages/store/utils/type.ts:29]: Do we still need this? If not, remove it

File: packages/store/migrations/index.ts (hunk 2 above)
Comment (tanlethanh) [reply packages/store/migrations/index.ts:69]: yes, it need to return a list of promise to use `await Promise.all()`

**Review by FME849 — APPROVED** [review]

**Pull request merged**